// components/Background.jsx
import React, {Component, PropTypes, ReactCSSTransitionGroup} from 'react/addons'
import {Fetcher} from 'utils'
import {Img} from 'components'

const TIME_TO_SHOW_IMAGE = 50000; //ms

// TODO ReactCSSTransitionGroup tem um bug ao trocar de aba no browser; nao ha patch ainda
class Background extends Component {
    propTypes = {
        images: PropTypes.array.isRequired
    }

    getInitialState = () => {imageIndex: 0}

    componentDidMount = () => this.interval = setInterval(this._gotoNextImage, TIME_TO_SHOW_IMAGE)
    componentWillUnmount = () => clearInterval(this.interval)

    _gotoNextImage = () => {
        var nextIndex = (this.state.imageIndex + 1) % this.props.images.length;
        this.setState({imageIndex: nextIndex});
    }

    render = (styles = this.styles, imgSrc = this.props.images[this.state.imageIndex]) =>
        <div style={styles.container}>
            <ReactCSSTransitionGroup transitionName="carousel">
                <Img style={styles.img} src={imgSrc} key={imgSrc}/>
            </ReactCSSTransitionGroup>
        </div>

    styles = {
        container: {
            height: '100%',
            left: 0,
            position: 'fixed',
            top: 0,
            width: '100%',
            zIndex: -999
        },

        img: {
            position: 'fixed',
            width: '100%'
        }
    }
}




export default Fetcher.wrap(Background)
