// components/ItemList.js
var React = require('react');
var PropTypes = React.PropTypes;
var ctx = require('utils/ctx');

var Img = require('./Img');
var LightBox = require('./LightBox');

var ItemList = React.createClass({
    _openLightbox(index) {
        this.refs.lightbox.open(index);
    },

    render() {
        var lightboxItems = [];
        var items = [];

        // TODO recuperar w e h do nome do item: "/item/image_50x50.jpg"
        this.props.items.forEach(function (item, i) {
            lightboxItems.push({
                src: ctx() + "/" + item,
                w: 400,
                h: 600
            });

            items.push(
                <div style={_.item} key={i} onClick={this._openLightbox.bind(this, i)}>
                    <Img src={item}/>
                </div>
            )
        }, this);

        return (
            <div style={_.container}>
                {items}

                <LightBox ref="lightbox" items={lightboxItems}/>
            </div>
        );
    }
});

var _ = {
    container: {},
    item: {
        cursor: 'pointer',
        display: 'inline-block',
        margin: 10
    }
};

module.exports = require('utils/Fetcher').wrap(ItemList);
