// components/LightBox.js
var React = require('react');
var PropTypes = React.PropTypes;

var PhotoSwipe = require('photoswipe');
var PhotoSwipeUI_Default = require('photoswipe/dist/photoswipe-ui-default');

var LightBox = React.createClass({

    _open() {
        var pswpElement = React.findDOMNode(this.refs.pswp);

        var items = [
            {
                src: 'img/bg/1425219857.jpg',
                w: 800,
                h: 600
            },
            {
                src: 'img/cat1/1410522050_tb.jpg',
                w: 200,
                h: 300
            },
            {
                src: 'img/cat1/1410522069_tb.jpg',
                w: 200,
                h: 300
            },
            {
                src: 'img/cat1/1410522089_tb.jpg',
                w: 200,
                h: 300
            },

            {
                src: 'img/cat2/1391627539_1_tb.jpg',
                w: 200,
                h: 300
            },
            {
                src: 'img/cat2/1391627539_2_tb.jpg',
                w: 200,
                h: 300
            },
            {
                src: 'img/cat2/1391627539_3_tb.jpg',
                w: 200,
                h: 300
            },
            {
                src: 'img/cat2/1391627539_tb.jpg',
                w: 200,
                h: 300
            },

        ];

        var options = {
            index: 0, // start at first slide,
            history: false,
            closeOnScroll: false
        };

        var gallery = new PhotoSwipe(pswpElement, PhotoSwipeUI_Default, items, options);
        gallery.init();
    },

    render() {
        return (
            <div>
                <button type="button" onClick={this._open}>Open Lightbox</button>

                <div ref="pswp" className="pswp" tabIndex="-1" role="dialog" aria-hidden="true">
                    <div className="pswp__bg"></div>

                    <div className="pswp__scroll-wrap">

                        <div className="pswp__container">
                            <div className="pswp__item"></div>
                            <div className="pswp__item"></div>
                            <div className="pswp__item"></div>
                        </div>

                        <div className="pswp__ui pswp__ui--hidden">

                            <div className="pswp__top-bar">
                                <div className="pswp__counter"></div>

                                <button className="pswp__button pswp__button--close" title="Close (Esc)"></button>

                                <button className="pswp__button pswp__button--share" title="Share"></button>

                                <button className="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                                <button className="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                                <div className="pswp__preloader">
                                    <div className="pswp__preloader__icn">
                                        <div className="pswp__preloader__cut">
                                            <div className="pswp__preloader__donut"></div>
                                        </div>
                                    </div>
                                </div>
                            </div>

                            <div className="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                                <div className="pswp__share-tooltip"></div>
                            </div>

                            <button className="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
                            </button>

                            <button className="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
                            </button>

                            <div className="pswp__caption">
                                <div className="pswp__caption__center"></div>
                            </div>

                        </div>

                    </div>

                </div>
            </div>
        );
    }
});

var _ = {
    container: {}
};

module.exports = LightBox;
