// components/Menu.jsx
import React, {Component, PropTypes} from 'react'
import {Fetcher} from 'utils'

const CSSTransitionGroup = React.addons.CSSTransitionGroup

class Menu extends Component {
    static propTypes: {
        categories: PropTypes.array.isRequired
    }

    componentWillMount = () => console.log("%cMenu:componentWillMount", "color:red")

    render = ({categories} = this.props) => {
        console.log("Menu:render", this.state);

        return (
            <div id="menu" className="  no-text-select" >
                {categories.map((category, i) =>
                    category.children ?
                        <DropdownMenu item={category} key={i} /> :
                        <LinkMenu item={category} key={i} />
                )}
            </div>
        )
    }
}

class DropdownMenu extends Component {
    state = {open: false}

    toggle = () => this.setState({open: !this.state.open})

    render = ({item} = this.props) =>
        <div className="menu-item">
            <div className="menu-title" onClick={this.toggle}>
                {item.name}
            </div>
            <CSSTransitionGroup transitionName="carousel">
                {this.state.open &&
                <SubMenus items={item.children}/>}
            </CSSTransitionGroup>
        </div>
}

class SubMenus extends Component {
    render = ({items} = this.props) =>
        <div className="menu-sub">
            {items.map((item, i) =>
                <LinkMenu item={item} key={i}/>
            )}
        </div>
}

class LinkMenu extends Component {
    render = ({item} = this.props) =>
        <a className="menu-title"  href={"/items/category/" + item.id}>{item.name}</a>
}

// estilos em app.css

export default Fetcher.wrap(Menu)
