// admin/CategoriesTree.jsx
import React, { PropTypes, Component } from 'react'
import CategoryService from 'services/CategoryService'

class CategoriesTree extends Component {
    state = {
        categories: []
    }

    componentDidMount() {
        // this.setState({
        //     categories: CategoryService.list()
        // })
    }

    addCategory = (category) => {
        this.state.categories.push({name: category})
        this.setState({
            categories: this.state.categories
        });
    }

    render = ({categories} = this.state) =>
        <div style={s.tree}>
            {categories.map((category, i) =>
                <Card category={category.name}/>
            )}

            <CardAdd onSave={this.addCategory}/>
        </div>
}

const Card = ({category}) =>
    <div style={s.card}>
        {category}
    </div>

class CardAdd extends Component {
    state = {
        formMode: false
    }

    gotoFormMode = () => this.setState({formMode: true})

    leaveFormMode = () => this.setState({formMode: false})

    render = ({onSave} = this.props, {formMode} = this.state) =>
        <div>
            {formMode ?
                <CardAddForm onSave={onSave} onCancel={this.leaveFormMode}/>:
                <CardAddButton onClick={this.gotoFormMode}/>
            }
        </div>
}

class CardAddButton extends Component {

    render = ({onClick} = this.props) =>
        <div style={s.card} onClick={onClick}>
            +
        </div>
}

class CardAddForm extends Component {
    onSave = () => {
        this.props.onSave(this.refs.category.value)
        this.props.onCancel();
    }

    render = ({onCancel} = this.props) =>
        <div style={s.form}>
            <input ref="category" type="text" className="form-control" autoFocus={true}/>

            <button type="button" className="btn btn-primary" onClick={this.onSave}>
                Save
            </button>
            <button type="button" className="btn" onClick={onCancel}>
                Cancel
            </button>
        </div>
}


const s = {
    tree: {},

    card: {
        border: '1px solid #666',
        cursor: 'pointer',
        height: 30,
        lineHeight: '30px',
        marginBottom: 20,
        textAlign: 'center'
    },

    form: {}
}

export default CategoriesTree
